### ISSUE TITLE
Large ontology file fails to load

### ISSUE THREAD

**Issue by mmisw** [mmisw/mmiorr#153]
_From [paul.r.a...@gmail.com](https://code.google.com/u/104064418752273474769/) on July 08, 2009 14:27:09_

What steps will reproduce the problem? 1. Download the ontology located here: http://marinemetadata.org/2005/08/bodc-param 2. Attempt to upload an ontology at the portal by choosing the file and
clicking the "Load" button
3. After ~30 seconds, the following error is shown:
"Error reading file. Make sure it is an RDF file.
Server reports:

Could not parse uploadResults."

This file is large (7424 KB) and it doesn't seem like there is enough time
for it to finish uploading before the error message appears.

_Original issue: http://code.google.com/p/mmisw/issues/detail?id=153_

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on July 13, 2009 11:00:58_

Diagnosed:
- There is a maximum size of 2 MB for any "upload" operation.
- Error message should be consistent (something like "your ontology file is larger
than the default maximum size. Please let us know about this issue so we can help you
with the registration process.")

Will do the following:
- increment max size to 12 MB (which looks like an acceptable default)
- detect the event of exceeded size to properly report the error

**Status:** Diagnosed  
  **Owner:** carueda

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on July 24, 2009 10:41:34_

Max file size set to 200MB.
Please retry registering the ontology.

Will see how to detect the event of exceeded file size.

**Summary:** Large ontology file fails to load  
  **Labels:** ontmd

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on July 24, 2009 11:10:09_

On second thought, I'm closing this entry as Fixed.

If necessary, please create a new issue related with the proper error report.

**Status:** Fixed

**Comment by mmisw** [comment]
_From [paul.r.a...@gmail.com](https://code.google.com/u/104064418752273474769/) on July 24, 2009 11:21:56_

We may want to put a small notice of max upload size, if I were uploading a 250MB 
ontology and it didn't fail or prevent me from uploading until 200MB in, I would be 
frustrated.

Dealing with uploaded files is still tricky, unfortunately. The server won't know the 
complete file size unless you do some workaround to get it before the upload starts...

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on July 24, 2009 11:51:05_

Max size now set to unlimited.  Please test (and then verify or reject).
  
(The apache commons fileupload API doc is not very clear in this aspect --or I'm not
reading it well enough.)

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on July 27, 2009 09:08:56_

I did a test with the bodc-param ontology: the upload operation (not to confuse with
the register operation) completes OK.  (I stopped my test here, I didn't register the
ontology.)

BUT NOTE: This big file slows down my firefox very significantly! This is because the
ontology is kept all in memory. See new issue #156
